Sources/DeepLearning/Layer.swift

```diff
@@ -1169,6 +1169,34 @@ public struct UpSampling2D<Scalar: TensorFlowFloatingPoint>: Layer {
     }
 }
 
+/// An upsampling layer for 3-D inputs.
+@_fixed_layout
+public struct UpSampling3D<Scalar: TensorFlowFloatingPoint>: Layer {
+    @noDerivative public let size: Int
+
+    /// Creates an upsampling layer.
+    ///
+    /// - Parameter size: The upsampling factor for rows and columns.
+    public init(size: Int) {
+       self.size = size
+    }
+
+    /// Returns the output obtained from applying the layer to the given input.
+    ///
+    /// - Parameter input: The input to the layer.
+    /// - Returns: The output.
+    @differentiable
+    public func call(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
+        let shape = input.shape
+        let (batchSize, height, width, breadth, channels) = (shape[0], shape[1], shape[2]
+                                                            , shape[3], shape[4])
+        let scaleOnes = Tensor<Scalar>(ones: [1, 1, size, 1, size, 1, size, 1])
+        let upSampling = input.reshaped(to: [batchSize, height, 1, width, 1,
+                                            breadth, 1, channels]) * scaleOnes
```

Comment:
small question: `breadth` or `depth`?

Reply:
Actually depth and height would be similar. i think it should be length, width and height, i took your suggestion for now. Thank you.